client/src/components/navigation/Sidebar.tsx

```diff
@@ -0,0 +1,312 @@
+import React, { useState } from "react";
+
+import { useEffect } from "react";
+import { Link, useNavigate } from "react-router-dom";
+
+import { Box, Stack, SwipeableDrawer, ListItemButton, Avatar, Typography, Collapse, List } from "@mui/material";
+import { alpha } from "@mui/material/styles";
+import { SvgIconComponent, ExpandLess, ExpandMore, ExitToApp } from "@mui/icons-material";
+
+import { Permission, authReducerActionType } from "../../types/globalTypes";
+import api from "../../utils/api";
+import { useAuth } from "../../hooks/useAuth";
+import { usePathname } from "../../hooks/usePathname";
+import { useResponsive } from "../../hooks/useResponsive";
+
+import JBTLogoBlack from "../../assets/jbt-logo-black.png";
+import Scrollbar from "./Scrollbar";
+import navConfig from "./navConfig";
+
+interface NavItemProps {
+  item: {
+    title: string;
+    path?: string;
+    icon: SvgIconComponent;
+    children?: {
+      title: string;
+      path: string;
+      permissions?: Permission[];
+    }[];
+  };
+}
+
+interface NavItemChildProps {
+  path: string;
+  title: string;
+  permissions?: Permission[];
+}
+
+interface NavProps {
+  openDrawer: boolean;
+  onCloseDrawer: () => void;
+  onOpenDrawer: () => void;
+}
+
+/**
+ * The NavItem-Component displays a single item in the sidebar
+ * For Each Item we render a ListItemButton with an icon and a title
+ * If the item has children, we render a collapse component with a list of children
+ * @param item the item to be displayed
+ */
+function NavItem({ item }: NavItemProps) {
```

Comment:
We use the arrow function format for functions (() => {...}).


```suggestion
const NavItem = ({ item }: NavItemProps) => {
```